internal/worker/removal/worker.go

```diff
@@ -37,23 +54,32 @@ func (config Config) Validate() error {
 type removalWorker struct {
 	catacomb catacomb.Catacomb
 
-	cfg Config
+	cfg    Config
+	runner *worker.Runner
```

Comment:
Could we implement a `Report` method on the `removalWorker`, calling on to the runner's `Report` method?